File: 228. Summary Ranges/228. Summary Ranges.js
/**
 * @param {number[]} nums
 * @return {string[]}
 */
var summaryRanges = function (nums) {
  if (nums.length == 0) {
    return [];
  }

  const summaryRanges = [];
  let start = 0;

  for (let i = 1; i < nums.length + 1; i++) {
    if (i == nums.length || nums[i] !== nums[i - 1] + 1) {
      if (start === i - 1) {
        summaryRanges.push(`${nums[start]}`);
      } else {
        summaryRanges.push(`${nums[start]}->${nums[i - 1]}`);
      }

      start = i;
    }
  }

  return summaryRanges;
};

File: 228. Summary Ranges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        n = len(nums)

        if n == 0:
            return []

        ranges = []
        start = nums[0]
        for i in range(1, n):
            if nums[i] != nums[i - 1] + 1:
                if nums[i - 1] == start:
                    ranges.append(str(start))
                else:
                    ranges.append(f"{start}->{nums[i - 1]}")
                start = nums[i]

        if start == nums[n - 1]:
            ranges.append(str(start))
        else:
            ranges.append(f"{start}->{nums[n - 1]}")

        return ranges
